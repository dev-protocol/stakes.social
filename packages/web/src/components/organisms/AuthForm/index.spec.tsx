import React from 'react'
import { render } from '@testing-library/react'
import { AuthForm } from '.'
import 'src/__mocks__/window/matchMedia.mock'
import { useEffectAsync, useNetworkInRouter } from 'src/fixtures/utility'

jest.mock('src/fixtures/utility')
jest.mock('src/fixtures/dev-kit/hooks')

describe(`${AuthForm.name}`, () => {
  test('Snapshot', () => {
    ;(useNetworkInRouter as jest.Mock).mockImplementation(() => ({
      requestedChain: 'ethereum'
    }))
    ;(useEffectAsync as jest.Mock).mockImplementation(() => {})
    const component = render(<AuthForm market="market" />)
    const tree = component.baseElement
    expect(tree).toMatchSnapshot()
  })
})
