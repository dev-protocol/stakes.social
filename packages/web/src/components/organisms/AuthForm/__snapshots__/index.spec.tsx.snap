// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm Snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-eCImPb hOeXCi"
    >
      <form
        class="ant-form ant-form-horizontal"
        id="basic"
      >
        <div
          class="sc-jRQBWg ffKtpf"
        >
          <div
            class="sc-gKclnd evpXwD"
          >
            Project:
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-typography ant-typography-secondary"
                  >
                    (Not provide this feature yet on L2)
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-jRQBWg ffKtpf"
        >
          <div
            class="sc-gKclnd evpXwD"
          >
            Project name:
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-row ant-form-item sc-bdvvtL kPCGkR"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="sc-gsDKAQ bHMKjG"
                          >
                            <span
                              aria-label="deployment-unit"
                              class="anticon anticon-deployment-unit"
                              role="img"
                              style="color: grey;"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="deployment-unit"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M888.3 693.2c-42.5-24.6-94.3-18-129.2 12.8l-53-30.7V523.6c0-15.7-8.4-30.3-22-38.1l-136-78.3v-67.1c44.2-15 76-56.8 76-106.1 0-61.9-50.1-112-112-112s-112 50.1-112 112c0 49.3 31.8 91.1 76 106.1v67.1l-136 78.3c-13.6 7.8-22 22.4-22 38.1v151.6l-53 30.7c-34.9-30.8-86.8-37.4-129.2-12.8-53.5 31-71.7 99.4-41 152.9 30.8 53.5 98.9 71.9 152.2 41 42.5-24.6 62.7-73 53.6-118.8l48.7-28.3 140.6 81c6.8 3.9 14.4 5.9 22 5.9s15.2-2 22-5.9L674.5 740l48.7 28.3c-9.1 45.7 11.2 94.2 53.6 118.8 53.3 30.9 121.5 12.6 152.2-41 30.8-53.6 12.6-122-40.7-152.9zm-673 138.4a47.6 47.6 0 01-65.2-17.6c-13.2-22.9-5.4-52.3 17.5-65.5a47.6 47.6 0 0165.2 17.6c13.2 22.9 5.4 52.3-17.5 65.5zM522 463.8zM464 234a48.01 48.01 0 0196 0 48.01 48.01 0 01-96 0zm170 446.2l-122 70.3-122-70.3V539.8l122-70.3 122 70.3v140.4zm239.9 133.9c-13.2 22.9-42.4 30.8-65.2 17.6-22.8-13.2-30.7-42.6-17.5-65.5s42.4-30.8 65.2-17.6c22.9 13.2 30.7 42.5 17.5 65.5z"
                                />
                              </svg>
                            </span>
                          </div>
                          <input
                            class="sc-hKwDye bghYKm"
                            id="basic_projectName"
                            placeholder="OSS Project name"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-jRQBWg ffKtpf"
        >
          <div
            class="sc-gKclnd evpXwD"
          >
            Personal Access Token:
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-row ant-form-item sc-bdvvtL kPCGkR"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="sc-gsDKAQ bHMKjG"
                          >
                            <span
                              aria-label="code"
                              class="anticon anticon-code"
                              role="img"
                              style="color: grey;"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="code"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 00308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 00-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
                                />
                              </svg>
                            </span>
                          </div>
                          <input
                            class="sc-hKwDye bghYKm"
                            id="basic_personalAccessToken"
                            placeholder="Personal Access Token"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="display: flex; align-items: center; margin-bottom: 1em;"
        >
          <div
            style="display: flex; align-items: center; align-self: flex-end;"
          >
            <div
              class="sc-pVTFL eIuwjz"
            >
              <span
                aria-label="info-circle"
                class="anticon anticon-info-circle"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="info-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  />
                  <path
                    d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
            </div>
            <div
              style="font-size: 0.9em;"
            >
              <div>
                Please
                 
                <a
                  href="https://github.com/settings/tokens/new"
                  rel="noreferrer"
                  target="_blank"
                >
                  <span
                    style="word-break: normal;"
                  >
                    create a Personal Access Token without any scopes.
                  </span>
                </a>
              </div>
              <div>
                <span
                  style="word-break: normal;"
                >
                  The PAT is confidentially authenticated using the Khaos oracle(
                </span>
                <a
                  href="https://github.com/dev-protocol/khaos"
                  rel="noreferrer"
                  target="_blank"
                >
                  *
                </a>
                ).
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-jRQBWg ffKtpf"
        >
          <div
            style="display: flex; grid-column: 1/-1; justify-content: flex-end;"
          >
            <div
              class="sc-iCfMLu kA-DLBd"
            >
              <button
                class="sc-furwcr dWvzTC"
                type="submit"
              >
                Authenticate
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</body>
`;
